File: skellig-test-step-reader/src/main/java/org/skellig/teststep/reader/exception/ValidationException.java
package org.skellig.teststep.reader.exception;

public class ValidationException extends RuntimeException {

    private String testStepId;

    public ValidationException(Exception e) {
        super(e);
    }

    public ValidationException(String message) {
        super(message);
    }

    public ValidationException(String message, String testStepId) {
        this(message);
        this.testStepId = testStepId;
    }

    public ValidationException(String message, Throwable throwable) {
        super(message, throwable);
    }

    public String getTestStepId() {
        return testStepId;
    }
}

File: skellig-test-step-processing/src/main/kotlin/org/skellig/teststep/processing/exception/ValidationException.kt
package org.skellig.teststep.processing.exception

/**
 * Exception thrown when a validation error occurs when checking [validation details][org.skellig.teststep.processing.model.DefaultTestStep.validationDetails].
 * of a test step.
 *
 * @param message The error message describing the validation failure.
 * @param cause The cause of the validation failure (optional).
 */
class ValidationException(message: String?, cause: Throwable?) : RuntimeException(message, cause) {

    constructor(message: String?) : this(message, null)

    constructor(e: Exception?) : this("", e)

}